src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1717,15 +1717,15 @@ public void Warning(string p, bool isPersistent = false)
                 State = ElementState.PersistentWarning;
                 if (!string.Equals(persistentWarning, p))
                 {
-                    persistentWarning += p;
+                    persistentWarning += persistentWarning == "" ? p : $"{Environment.NewLine}{p}";
```

Comment:
We can probably use the same check here from below: `string.IsNullOrEmpty(persistentWarning)`